server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -561,12 +558,13 @@ public ClusterMetadataManifest markLastStateAsCommitted(ClusterState clusterStat
             previousManifest.getGlobalMetadataFileName(),
             true
         );
-        deleteStaleClusterUUIDs(clusterState, committedManifest);
+        remoteClusterStateCleanupManager.deleteStaleClusterUUIDs(clusterState, committedManifest);
```

Comment:
This should be optimized to not run for every cluster state update.
We can add condition that in previous manifest clusterUUIDCommitted was false and in current state clusterUUIDCommitted is true.

Reply:
This is codeflow is only triggered when we mark last state committed, i.e. when cluster state update cluster uuid committed from false to true. Am I missing something?

Reply:
There are two flags in the manifest - one for marking cluster state as committed and one for setting cluster uuid as committed. 
The above code flow is triggered everytime before the commit phase of cluster state publication for marking cluster state as committed..